static/js/publisher/release/actions/pendingReleases.js

```diff
@@ -70,17 +70,25 @@ export function promoteRevision(revision, channel) {
       );
     });
 
-    const canBeProgressive = revision.architectures.some(arch =>
-      hasRelease(getState(), channel, arch)
-    );
+    const previousRevisions = {};
+
+    revision.architectures.forEach(arch => {
```

Comment:
should we `const { channelMap } = getState()` this, so we don't need to call `getState()` twice - not sure how performant the function is in redux?

Reply:
Good call, updated.